File: src/main/java/it/contrader/dto/ProgressDTO.java
package it.contrader.dto;

import java.sql.Timestamp;



import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor

public class ProgressDTO {

		private Long id;
		
		private float cash;
		
		private double expectation;
		
		private double time;
		
		private ProdottoDTO prodotto; // creo una nuova variabile di tipo prodotto
		
		private Timestamp data;
}

File: src/dto/progressdto.ts
import  { ProdottoDTO } from './prodottodto'



export class ProgressDTO{

    id: number;

    cash: number;

    expectation: number;

    time: number;

    prodotto_idId: number;
    
    data : Date;

    prodotto: ProdottoDTO;

    constructor(cash?:number,expectation?:number,time?:number,prodotto_idId?:number, data?:Date, prodotto?: ProdottoDTO){

this.time=time;
this.expectation=expectation;
this.cash=cash;
this.prodotto_idId=prodotto_idId;
this.data=data;
this.prodotto = prodotto;

    }

}